openquake/calculators/event_based.py

```diff
@@ -294,6 +295,25 @@ def todict(dframe):
     return {k: dframe[k].to_numpy() for k in dframe.columns}
 
 
+def filter_stations(station_df, complete, hypo, maxdist):
+    """
+    :param station_df: DataFrame with the stations
+    :param complete: complete SiteCollection
+    :param hypo: hypocenter of the rupture
+    :param maxdist: maximum distance
+    :returns: filtered (station_df, station_sitecol)
+    """
+    ns = len(station_df)
+    ok = (hypo.distance_to_mesh(complete.mesh) <= maxdist) & numpy.isin(
```

Comment:
Filtering of the stations should be done using `rrup` as the distance measure, otherwise for large ruptures,  if we use `rhypo` then stations that are far from the hypocenter but still very close to the rupture itself can get excluded. 


![turkey](https://github.com/gem/oq-engine/assets/1707414/cb19ffc4-a6e8-4a59-b124-5c2240810a3f)